File: app/src/main/java/com/antyzero/smoksmog/user/UserModule.java
package com.antyzero.smoksmog.user;


import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module @Singleton
public class UserModule {

    @Provides @Singleton
    public User provideUser(Context context){
        return new User(context);
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/user/UserModule.kt
package com.antyzero.smoksmog.user


import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
@Singleton
class UserModule {

    @Provides
    @Singleton
    internal fun provideUser(context: Context): User = User(context)
}
